apps/examples/messaging_sample/messaging_main.c

```diff
@@ -0,0 +1,226 @@
+/****************************************************************************
+ * examples/messaging_sample/messaging_main.c
+ *
+ *   Copyright (C) 2008, 2011-2012 Gregory Nutt. All rights reserved.
+ *   Author: Gregory Nutt <gnutt@nuttx.org>
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name NuttX nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+#include <sched.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <messaging/messaging.h>
+
+#define TEST1_PORT "test1_port"
+#define TEST1_DATA "TEST1"
+
+#define TEST2_PORT "test2_port"
+#define TEST2_DATA "TEST2"
+#define TEST2_REPLY "REPLY_MSG"
+
+#define MSG_PRIO 10
+#define TASK_PRIO 100
+#define STACKSIZE 2048
+#define BUFFER_SIZE 100
+
+static void recv_callback(void *data, void *recv_data)
+{
+	printf("Success to receive with non-block mode in recv callback. : %s\n", (char *)recv_data);
+}
+
+int async_send(int argc, FAR char *argv[])
+{
+	int ret;
+	msg_send_param_t param;
+	param.priority = MSG_PRIO;
+	param.type = MSG_SEND_TYPE_ASYNC;
+	param.msg = TEST1_DATA;
+	param.msglen = sizeof(TEST1_DATA);
+
+	printf("- Send %s data with async mode.\n", TEST1_DATA);
+	ret = messaging_send(TEST1_PORT, &param, NULL);
+	if (ret != OK) {
+		printf("Fail to async send message.\n");
+	}
+
+	return 0;
```

Comment:
always return 0, then void is better.
Some of tools will treat this as a defect(minor level)

Reply:
Yes, but I matched the task_create param type.

Reply:
I added another return for ERROR. Thanks.